executor/src/witgen/jit/witgen_inference.rs

```diff
@@ -145,21 +145,21 @@ impl<'a, T: FieldElement, FixedEval: FixedEvaluator<T>> WitgenInference<'a, T, F
     }
 
     /// Process an identity on a certain row.
-    /// Returns true if there was progress.
+    /// Returns Ok(true) if there was progress and Ok(false) if there was no progress.
+    /// If this returns an error, it means we have conflicting constraints.
     pub fn process_identity<CanProcess: CanProcessCall<T>>(
         &mut self,
         can_process: CanProcess,
         id: &'a Identity<T>,
         row_offset: i32,
-    ) -> bool {
-        // TODO remove this once we propagate range constraints.
-        if self.is_complete(id, row_offset) {
```

Comment:
Could you add something to the docstring of `complete_identities`, like "Completed identities can still be processed to propagate concrete values"?

Reply:
Also, to understand this better, we still process machine calls (we call `can_process_call_fully`), we just don't write it into the generated code, right? So, we would be able to transfer a concrete value from `a` to `b` here (assuming `can_process_call_fully` is implemented well):
`[a, b] in [c, c]`